test/Microsoft.AspNet.Mvc.Core.Test/RemoteAttributeTest.cs

```diff
@@ -0,0 +1,799 @@
+                Routers =
+                {
+                    routeCollection,
+                },
+                Values =
+                {
+                    { "action", "Index" },
+                    { "controller", "Home" },
+                },
+            };
+            if (!string.IsNullOrEmpty(currentArea))
+            {
+                routeData.Values["area"] = currentArea;
+            }
+
+            var contextAccessor = GetContextAccessor(serviceProvider, routeData);
+            var actionSelector = new Mock<IActionSelector>(MockBehavior.Strict);
+            var urlHelper = new UrlHelper(contextAccessor, actionSelector.Object);
+            serviceCollection.AddInstance<IUrlHelper>(urlHelper);
+            serviceProvider = serviceCollection.BuildServiceProvider();
+
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var metadata = metadataProvider.GetMetadataForProperty(
+                modelAccessor: null,
+                containerType: typeof(string),
+                propertyName: "Length");
+
+            return new ClientModelValidationContext(metadata, metadataProvider, serviceProvider);
+        }
+
+        private static ClientModelValidationContext GetValidationContextWithNoController()
+        {
+            var serviceCollection = GetServiceCollection();
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            var routeCollection = GetRouteCollectionWithNoController(serviceProvider);
+            var routeData = new RouteData
+            {
+                Routers =
+                {
+                    routeCollection,
+                },
+            };
+
+            var contextAccessor = GetContextAccessor(serviceProvider, routeData);
+            var actionSelector = new Mock<IActionSelector>(MockBehavior.Strict);
+            var urlHelper = new UrlHelper(contextAccessor, actionSelector.Object);
+            serviceCollection.AddInstance<IUrlHelper>(urlHelper);
+            serviceProvider = serviceCollection.BuildServiceProvider();
+
+            var metadataProvider = new EmptyModelMetadataProvider();
+            var metadata = metadataProvider.GetMetadataForProperty(
+                modelAccessor: null,
+                containerType: typeof(string),
+                propertyName: "Length");
+
+            return new ClientModelValidationContext(metadata, metadataProvider, serviceProvider);
+        }
+
+        private static IRouter GetRouteCollectionWithArea(IServiceProvider serviceProvider)
+        {
+            var builder = new RouteBuilder
+            {
+                ServiceProvider = serviceProvider,
+            };
+
+            var handler = new Mock<IRouter>(MockBehavior.Strict);
+            handler
+                .Setup(e => e.GetVirtualPath(It.IsAny<VirtualPathContext>()))
+                .Callback<VirtualPathContext>(context =>
+                {
+                    builder.ToString();
```

Comment:
looks like the Moq / Roslyn issue ([#246](https://roslyn.codeplex.com/workitem/246)) has been resolved enough that we don't need to force a closure here.